### ISSUE TITLE
[HOLD for payment 2024-02-20] [$1000] [Polish] Add limit on length of task names, room names, etc. 

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#18647]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any Expensifail account
3. Go to FAB -> Assign task
4. Start typing the title and note that there is no symbol limit

## Expected Result:
After discussion in Slack, we want to make the following field limit updates: 

**Titles (100 characters):**

- Workspace name
- Task name
- Room name

**Supporting messages (500 characters):**

- Request description
- Room welcome message
- Task description

## Actual Result:
There is no limit on the length of the title.
Same behavior for Description

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.12.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/05b9957a-b07e-4625-9e73-3b720dea6b93

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018a42c5fcacc50150</li>
        <li>Upwork Job ID: 1686014963280871424</li>
        <li>Last Price Increase: 2023-07-31</li>
<li>Automatic offers: </li>
<ul>
<li>dhairyasenjaliya | Contributor | 26010517</li>
</ul></li>
    </ul>
</details>

**Comment by joekaufmanexpensify** [comment]
My initial thought is this is not a bug, and should be a new feature issue. Confirming [here. ](https://expensify.slack.com/archives/C04QEB4MJEQ/p1683651648408999)

**Comment by joekaufmanexpensify** [comment]
Still getting confirmation that this should be a new feature issue, and not a bug.

**Comment by joekaufmanexpensify** [comment]
Okay, we discussed this in Slack [here](https://expensify.slack.com/archives/C04QEB4MJEQ/p1683745905041779?thread_ts=1683651648.408999&cid=C04QEB4MJEQ), and I confirmed this is not a bug. We just didn't build this. 

While it prob makes sense to add a limit at some point, we decided we don't need to prioritize this and it can be polish. Particularly because: 

- We also don't have a limit for room names, and this has caused no issues so far. 
- We don't allow multiple characters by holding down a key in the app. So if someone wants to make a task that's 100 characters long, they need to type 100 keys. It doesn't feel super realistic that a user would intentionally make a room or task name that's longer than any limit we decided to set. 

When we do ultimately decide to do this, we'll want to consider all of the areas that don't have a limit, like task name, room name, etc. and add a limit to all of them.

**Comment by joekaufmanexpensify** [comment]
Next steps here are to:

1. identify anywhere in the app that needs a name limit. 
2. Figure out what we think that limit should be. 
3. Implement the limit in all of those places. 

This is a minor polish issue, so will get to it in the next few weeks.

**Comment by joekaufmanexpensify** [comment]
Sounds good!

**Comment by joekaufmanexpensify** [comment]
Will prioritize this soon.

**Comment by joekaufmanexpensify** [comment]
OOO next week, but then will pick this back up.

**Comment by joekaufmanexpensify** [comment]
Was OOO last week, going to look into this soon

**Comment by joekaufmanexpensify** [comment]
Going to focus on this this week!

**Comment by joekaufmanexpensify** [comment]
Started a discussion with proposed limits [here.](https://expensify.slack.com/archives/C01GTK53T8Q/p1686863510616919)

**Comment by dangrous** [comment]
Do we have a plan for workspace name limits too? I saw https://expensify.slack.com/archives/C01GTK53T8Q/p1686952259007779?thread_ts=1686863510.616919&cid=C01GTK53T8Q but wasn't sure if workspace names would be included in one of those categories as well. Asking due to https://github.com/Expensify/App/issues/19359 - thanks!

**Comment by joekaufmanexpensify** [comment]
Did some research on why we selected existing limits last week. Going to bring to slack today!

**Comment by joekaufmanexpensify** [comment]
Ran out of time for this last week. But going to post an update today.

**Comment by joekaufmanexpensify** [comment]
I posted an update on this in the [slack discussion today](https://expensify.slack.com/archives/C01GTK53T8Q/p1688561105668269?thread_ts=1686863510.616919&cid=C01GTK53T8Q)!

**Comment by joekaufmanexpensify** [comment]
Still discussing in Slack thread after last week's update!

**Comment by joekaufmanexpensify** [comment]
We agreed on limit thresholds in Slack! Next step is for me to identify where we're going to change the limits this week

**Comment by joekaufmanexpensify** [comment]
Posting an update to the slack thread today!

**Comment by joekaufmanexpensify** [comment]
[Update here](https://expensify.slack.com/archives/C01GTK53T8Q/p1690209885486499?thread_ts=1686863510.616919&cid=C01GTK53T8Q)

**Comment by joekaufmanexpensify** [comment]
Updated the new expected result in OP based on where the slack discussion landed. We're not all set to work on this!

**Comment by joekaufmanexpensify** [comment]
cc @allroundexperts @dhairyasenjaliya if you want to update your previous proposals to match the next expected behavior here, we're now all set to proceed!

**Comment by mananjadhav** [comment]
Taking this over.

**Comment by mananjadhav** [comment]
@dhairyasenjaliya's proposal [here](https://github.com/Expensify/App/issues/18647#issuecomment-1658448685). I was wondering if we should just have two constants `TITLE_MAX_LENGTH` and `SUPPORTING_TEXT_MAX_LENGTH`.

Will also take feedback form an internal engineer.

🎀 👀 🎀 C+ reviewed

**Comment by mananjadhav** [comment]
Quick bump on this one @youssef-lr.

**Comment by joekaufmanexpensify** [comment]
Bumped Youssef 1:1.

**Comment by joekaufmanexpensify** [comment]
Youssef's been OOO, but is going to review this one soon!

**Comment by joekaufmanexpensify** [comment]
Still pending input from @youssef-lr

**Comment by youssef-lr** [comment]
LGTM @mananjadhav, we can decide on the details in the PR I think.

**Comment by mananjadhav** [comment]
@dhairyasenjaliya I think you can complete the PR.

@joekaufmanexpensify @youssef-lr Can you folks help with the backend change here?

**Comment by joekaufmanexpensify** [comment]
Sure! @youssef-lr since the auto-assigner selected you for the proposal review, do you want to handle the backend changes for the field limits too?

**Comment by youssef-lr** [comment]
Sure thing!

**Comment by joekaufmanexpensify** [comment]
Awesome, ty!

**Comment by joekaufmanexpensify** [comment]
Next step here is internal PR!

**Comment by mananjadhav** [comment]
@dhairyasenjaliya Still waiting for an update on this one.

**Comment by mananjadhav** [comment]
@dhairyasenjaliya As I've mentioned before we can finish the PR on the frontend. It isn't that this is introducing the bug currently? This anyway exists today, that backend has limit but we don't have on the frontend.

I don't see a need to block the App PR. Thanks.

**Comment by mananjadhav** [comment]
and what happens today if we enter the text of length 100?

**Comment by mananjadhav** [comment]
Okay got it. I thought we didn't have the limit on the frontend. Thanks for correcting me. @joekaufmanexpensify can we put this on Hold until we have the backend fix up?

**Comment by joekaufmanexpensify** [comment]
@mananjadhav should it be on hold though? We're going to handle the internal PR in this issue too.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr could you share an ETA for when you expect the internal PR to be up?

**Comment by mananjadhav** [comment]
Yeah it makes sense to put this on Hold. We could split them into separate PRs, but I don't think it's worth the effort as backend fix should be a small one. @youssef-lr You agree?

**Comment by joekaufmanexpensify** [comment]
Got it. I'd think the issue would not be on hold though. The frontend PR is held on the backend one. But the backend one is still being managed in this issue too. Could we just put that PR on hold?

I'm not sure it makes sense to hold the entire issue, as part of it can be worked on now.

**Comment by mananjadhav** [comment]
Okay sure I thought we create separate issues for backend. Thanks for correcting. Then it makes sense to put the PR on Hold. cc - @dhairyasenjaliya

**Comment by joekaufmanexpensify** [comment]
@youssef-lr could you share an ETA to the issue of when the internal PR will be up here?

**Comment by daordonez11** [comment]
Hey guys coming from https://github.com/Expensify/App/issues/25494, please include in the polish the Description and the merchant of the request. As per discussed here:
https://expensify.slack.com/archives/C01GTK53T8Q/p1692723237220349

cc @dhairyasenjaliya @mananjadhav @joekaufmanexpensify @mountiny

**Comment by joekaufmanexpensify** [comment]
@mananjadhav bumped the discussion [here. ](https://expensify.slack.com/archives/C02NK2DQWUX/p1692812064834929)

**Comment by joekaufmanexpensify** [comment]
Latest ETA for new PR is tomorrow!

**Comment by Santhosh-Sellavel** [comment]
We closed https://github.com/Expensify/App/issues/26123 in favor of this issue. It makes sense to handle it here, please take of it.

cc: @mananjadhav

**Comment by joekaufmanexpensify** [comment]
I bumped the slack discussion again.

**Comment by joekaufmanexpensify** [comment]
Still held on internal PR.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr could you please share an update on this when you have a chance? TY!

**Comment by youssef-lr** [comment]
@joekaufmanexpensify I'm gonna be able to get to this one after the launch of TU, which is in a couple of days, promise!

**Comment by joekaufmanexpensify** [comment]
Cool cool, sounds great, ty!

**Comment by joekaufmanexpensify** [comment]
We'll get to this one later this week, once TU is done

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
We're discussing some error handling considerations [here](https://github.com/Expensify/App/issues/25804#issuecomment-1705239139) that will impact how we proceed with fixing this issue.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr could you please confirm once you're ready to proceed with this one?

**Comment by youssef-lr** [comment]
Working on the PR @joekaufmanexpensify!

**Comment by joekaufmanexpensify** [comment]
Sounds good. We [are discussing](https://github.com/Expensify/App/issues/25804#issuecomment-1705239139) adding a character counter as an error message for when the user hits the limit, so they'd be able to see that they have exceeded the limit, but just not be cut off from typing. 

That would be something to incorporate into the frontend PR though, right?

**Comment by joekaufmanexpensify** [comment]
Still pending further discussion on the above point.

**Comment by youssef-lr** [comment]
>That would be something to incorporate into the frontend PR though, right?

Correct, this can be made external.

**Comment by joekaufmanexpensify** [comment]
Got it, ty!

**Comment by joekaufmanexpensify** [comment]
Not overdue

**Comment by joekaufmanexpensify** [comment]
Still ironing out specifics of character counter in the above linked issue.

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still discussing the character limit error patterns in the above thread.

**Comment by joekaufmanexpensify** [comment]
Making progress in the above linked discussion. Should be able to move forward here based on that soon.

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
[Bumped](https://github.com/Expensify/App/issues/25804#issuecomment-1735182736) the related discussion.

**Comment by joekaufmanexpensify** [comment]
Still pending further discussion

**Comment by joekaufmanexpensify** [comment]
Bumped the related discussion to C+ in Slack

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Making progress in the error handling discussion in the other issue. i think it's going to be resolved soon!

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Almost ready to proceed here!

**Comment by joekaufmanexpensify** [comment]
I think we are all set to move forward on the error handling discussions. Commented about this [here, ](https://github.com/Expensify/App/issues/25804#issuecomment-1751269515) and going to update this issue on Monday!

**Comment by trjExpensify** [comment]
Sounds good! CC: @luacmartins when this is done, the description field will be limited to 500 characters.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav @dhairyasenjaliya @youssef-lr Sorry for the delay here, but we are officially ready to proceed with this one. The only new thing we need to add on top of what we previously scoped is error handling/the character counter that we landed on in https://github.com/Expensify/App/issues/25804 

The specific behavior is described [here. ](https://github.com/Expensify/App/issues/25804#issuecomment-1743643174)

Can y'all take a look at that and LMK if you have any questions, if anything else is needed to pick this one back up? TY!

**Comment by joekaufmanexpensify** [comment]
Yep, what we landed on is listed in OP. We just need to add the [error handling](https://github.com/Expensify/App/issues/18647#issuecomment-1760444781) we standardized on in the other issue to all of those fields as well.

**Comment by joekaufmanexpensify** [comment]
I don't think @youssef-lr made the backend changes yet, right?

**Comment by mananjadhav** [comment]
Will wait for @youssef-lr to confirm if the changes are done on the backend.

**Comment by mananjadhav** [comment]
Quick bump @youssef-lr? @dhairyasenjaliya Are we unblocked with the PR here?

**Comment by dhairyasenjaliya** [comment]
yes currently limit for workspace name is 80 on backend and we need 100 as per latest update @mananjadhav @youssef-lr

**Comment by youssef-lr** [comment]
Will get to this this week folks.

**Comment by joekaufmanexpensify** [comment]
Cool cool, ty!

**Comment by joekaufmanexpensify** [comment]
@youssef-lr will you be able to raise the internal PR this week?

**Comment by youssef-lr** [comment]
@joekaufmanexpensify @mananjadhav so just the workspace name set to 80 max?

**Comment by youssef-lr** [comment]
Got it, will raise a PR this week.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr @dhairyasenjaliya Wouldn't we need an internal PR to update the max length of [all of the fields we're changing](https://github.com/Expensify/App/issues/18647#issue-1702316175) in this issue? Instead of just workspace name?

**Comment by mananjadhav** [comment]
Quick bump @youssef-lr.

**Comment by youssef-lr** [comment]
On it

**Comment by joekaufmanexpensify** [comment]
@youssef-lr is your PR updating the field limit lengths for the other fields mentioned in the [expected result](https://github.com/Expensify/App/issues/18647#issue-1702316175) section of OP too?

**Comment by joekaufmanexpensify** [comment]
Ah, interesting. How would we adjust that limit then?

**Comment by jaimishg2098** [comment]
### Proposal 

There is no validation condition in validate function and also max-length prop is not there in <InputWrapper />

**Comment by joekaufmanexpensify** [comment]
@youssef-lr [mind confirming](https://github.com/Expensify/App/issues/18647#issuecomment-1808956319) how we'd adjust the limit for all of these other fields?

**Comment by joekaufmanexpensify** [comment]
Pending further input

**Comment by joekaufmanexpensify** [comment]
bump @youssef-lr for when you have a sec

**Comment by joekaufmanexpensify** [comment]
Still pending further input

**Comment by joekaufmanexpensify** [comment]
Bumped Youssef 1:1

**Comment by joekaufmanexpensify** [comment]
@youssef-lr advised that he's going to work on this before EOW

**Comment by joekaufmanexpensify** [comment]
@youssef-lr just checking in to make sure we're still on track with this PR?

**Comment by youssef-lr** [comment]
Yes I'm on it today @joekaufmanexpensify!

**Comment by mananjadhav** [comment]
@youssef-lr Is backend part done? It's been open for quite sometime.

**Comment by mananjadhav** [comment]
I think we should also cover [this here](https://github.com/Expensify/App/issues/31874)

**Comment by youssef-lr** [comment]
Noted @mananjadhav , PR should be ready in a couple of hours.

**Comment by youssef-lr** [comment]
PR is ready for review

**Comment by joekaufmanexpensify** [comment]
@youssef-lr I see you added some new commits to the PR. Is it ready for re-review?

**Comment by joekaufmanexpensify** [comment]
PR still pending further review

**Comment by youssef-lr** [comment]
Pushing one last update @joekaufmanexpensify

**Comment by joekaufmanexpensify** [comment]
Sounds good!

**Comment by joekaufmanexpensify** [comment]
@youssef-lr I see the web PR is out on production. Are we good to start on the frontend changes?

**Comment by youssef-lr** [comment]
Correct @joekaufmanexpensify. I see this issue mentioned above me, I think we should create an issue to get these all fixed and close any other issue about an individual field

**Comment by dhairyasenjaliya** [comment]
hey @joekaufmanexpensify @youssef-lr if backend changes are done I can continue on PR and also address all the field at once

**Comment by youssef-lr** [comment]
Sounds good @dhairyasenjaliya, please do and pull me for review. Thanks!

**Comment by dhairyasenjaliya** [comment]
Cool will rise soon

**Comment by dhairyasenjaliya** [comment]
@youssef-lr @joekaufmanexpensify I have added validation as stated on OP please let me know if we need to add validation on other places as well

**Comment by joekaufmanexpensify** [comment]
Is all that's left here the above frontend PR?

**Comment by mananjadhav** [comment]
@youssef-lr I've added a comment on the PR can you take a look?

**Comment by joekaufmanexpensify** [comment]
@mananjadhav I see a recent review checklist on the PR. Is this still pending input? If so, I will DM Youssef 1:1 to get this.

**Comment by mananjadhav** [comment]
Yeah the change I am requesting isn't the blocker. I'll finish the testing today/tomorrow.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav approved the PR. @youssef-lr could you please review when you have a sec? TY!

**Comment by youssef-lr** [comment]
Replied in the PR!

**Comment by youssef-lr** [comment]
Reviewed, once conflicts are fixed and small comment is addressed we can merge.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr are we good to merge now?

**Comment by youssef-lr** [comment]
Still being reviewed @joekaufmanexpensify, just left another comment

**Comment by joekaufmanexpensify** [comment]
PR merged. Though I see there's a deploy blocker linked to it

**Comment by youssef-lr** [comment]
@joekaufmanexpensify it seems like it's not related to the PR.

**Comment by mananjadhav** [comment]
Yes I posted [here](https://github.com/Expensify/App/pull/33601#issuecomment-1938913405) and [here](https://github.com/Expensify/App/issues/36342#issuecomment-1938915156) as well along with the screenshot.

<img width="1512" alt="Screenshot 2024-02-12 at 9 06 41 PM" src="https://github.com/Expensify/App/assets/3069065/1153ef6a-4309-4515-85a0-2a3457817852">

**Comment by youssef-lr** [comment]
@dhairyasenjaliya can you please look into the linked issue above.

**Comment by dhairyasenjaliya** [comment]
hey @youssef-lr yes I think at the time of solving conflicts this might be left to remove limit I will create PR shortly

**Comment by dhairyasenjaliya** [comment]
@youssef-lr created PR above please check

**Comment by mananjadhav** [comment]
@dhairyasenjaliya Posted comment on the PR.

**Comment by dhairyasenjaliya** [comment]
@mananjadhav all done

**Comment by youssef-lr** [comment]
Merged, thanks folks. @joekaufmanexpensify for the payments, I don't think this should count as a regression.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav could you please propose regression test so we can prep to issue payment next week?

**Comment by joekaufmanexpensify** [comment]
Payment due soon

**Comment by mananjadhav** [comment]
@joekaufmanexpensify We have got multiple test cases for this one. I think we can reuse the ones mentioned in the [Test steps of the PR](https://github.com/Expensify/App/pull/33601#issue-2056509058).

There won't be any offending PR as this was more of a feature request.

**Comment by joekaufmanexpensify** [comment]
@dhairyasenjaliya $1000 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@mananjadhav could you please request $1000 via NewDot and confirm once complete?

**Comment by mananjadhav** [comment]
Done @joekaufmanexpensify. Thanks for the bump.

**Comment by joekaufmanexpensify** [comment]
Awesome. TY! Closing as this is all set. Thanks everyone!

**Comment by JmillsExpensify** [comment]
$1,000 approved for @mananjadhav based on summary above.
